### ISSUE TITLE
[switch] Should a switch proposal graduate to the web platform?

### ISSUE THREAD

**Issue by gregwhitworth** [openui/open-ui#338]
While [investigating the checkbox](https://open-ui.org/components/checkbox.research.findings) we found some sites had a toggle or switch state for their checkbox. This is not necessarily surprising as it can be seen here as well in the [switch](research concepts page).

I think to fully answer this question however there will need to be a bit more research as I'm truly asking from a graduation perspective. I do feel that based on the amount of systems that have a toggle or switch that a spec for a good switch component should be within Open UI. I do feel that that is a good argument for having on in the platform but there are certain push backs on this given how close the semantics of a checkbox & switch are.

I'm creating this to get others to weigh in and hopefully provide further research PRs into site to help inform the direction.

**Comment by una** [comment]
Toggles are very popular. They are often a component that a design system provides because there is no 1-to-1 styled mapping to existing HTML, so +1 I do think they're a good candidate for HTML 👍🏼

How should we track this between here and whatwg?

**Comment by chrisdholt** [comment]
Agree with @una - name can be discussed at a later date; my bent is to go with "switch" per the respective ARIA role and the fact that "toggle buttons" are also a concept but fundamentally different. Either way, I'm in agreement on moving this forward from a process standpoint.

**Comment by gregwhitworth** [comment]
I tend to agree as well. As it relates to graduation, @blittle and myself have worked out a game plan around `checkbox` and this seems like one that can piggy back on that one given how much overlap exists from a behavior perspective.

**Comment by gregwhitworth** [comment]
This was discussed in today's telecon with the following resolution:

> RESOLUTION: Open-UI should have a separate Switch blueprint proposal

[Minutes are here](https://www.w3.org/2021/05/13-openui-minutes.html)

**Comment by tabatkins** [comment]
> I shouldn't match the indetermittend select either

Indeterminate switches aren't too uncommon. With a light/dark theme switcher, for example, the default "system's choice" is sometimes represented with the switch in the middle, and that does match the "indeterminate" state reasonably well. However, it's currently impossible to revert a checkbox to indeterminate via user action, so that's awkward.
